annotator-core/src/test/java/edu/ucr/cs/riple/core/CoreTest.java

```diff
@@ -451,4 +451,41 @@ public void staticBlockLocalVariableInitializationTest() {
         .enableForceResolve()
         .start();
   }
+
+  @Test
+  public void impactedLambdaAndMemberReferenceTest() {
+    coreTestHelper
+        .addInputLines(
+            "Main.java",
+            "package test;",
+            "import javax.annotation.Nullable;",
+            "public class Main {",
+            "   public void f(Foo r){ }",
+            "   public void baz1(){",
+            "     f(new Foo(){",
+            "       @Override",
+            "       public void bar(Object o){",
+            "         take(o);",
+            "       }",
+            "     });",
+            "   }",
+            "   public void baz2() {",
+            "       f(e -> take(e));",
+            "   }",
+            "   public void baz3() {",
+            "       f(this::take);",
```

Comment:
Now that I understand better the method reference handling, can we somehow split this into two different tests for method references?  One test should check that if `@Nullable` is added to the parameter of the functional interface, the region enclosing the method reference is treated as impacted.  A second test should check that if `@Nullable` is added to the return type of the referenced method, the region enclosing the method reference is treated as impacted.  I think this line is somehow testing both simultaneously.  You may need to add a second functional interface to test each case separately.  With these separate tests, both the cases from our discussion about method references would be explicit in the tests

Reply:
Sure, this is a lot more clear now. I added a separate unit test that is checking the impact of making the called method in `method references` get annotated as `@Nullable` and the other one is checking the impact of making the parameter in the overridden method in the interface as `@Nullable`.  3d182ceececa0472274fb076f549731b22ca4bdd